server/models/dataset_helpers/image_metadata.py

```diff
@@ -515,4 +518,7 @@ def addImageClinicalMetadata(image, data):
     # Add remaining data as unstructured metadata
     image['meta']['unstructured'].update(data)
 
-    return errors
+    # Report unrecognized fields when there are no errors
+    unrecognizedFields = six.viewkeys(data) if not errors else None
+
+    return errors, unrecognizedFields
```

Comment:
Ideally, this would return a tuple of general `(errors, warnings)`, but this wouldn't work with the logic to only capture the first row's "warnings". I can't think of a good solution right now, though.

Reply:
I changed the logic at the model layer and changed this function to return `(errors, warnings)`.